aepsych/strategy.py

```diff
@@ -207,57 +207,63 @@ def normalize_inputs(
 
     # TODO: allow user to pass in generator options
     @ensure_model_is_fresh
-    def gen(self, num_points: int = 1):
+    def gen(self, num_points: int = 1) -> torch.Tensor:
         """Query next point(s) to run by optimizing the acquisition function.
 
         Args:
             num_points (int, optional): Number of points to query. Defaults to 1.
             Other arguments are forwared to underlying model.
 
         Returns:
-            np.ndarray: Next set of point(s) to evaluate, [num_points x dim].
+            torch.Tensor: Next set of point(s) to evaluate, [num_points x dim].
         """
         self._count = self._count + num_points
         return self.generator.gen(num_points, self.model)
 
     @ensure_model_is_fresh
-    def get_max(self, constraints=None, probability_space=False, max_time=None):
+    def get_max(self, constraints: Optional[Mapping[int, List[float]]]  = None, probability_space: bool = False, max_time: Optional[float] = None) -> Tuple[float, torch.Tensor]:
         constraints = constraints or {}
+        assert self.model is not None, "a model is needed here!"
```

Comment:
I would make this message, and all the similar ones, more descriptive. Like "model is None! Cannot get the max without a model!"